plugins/window-state/src/lib.rs

```diff
@@ -24,12 +24,15 @@ use std::{
     collections::{HashMap, HashSet},
     fs::{create_dir_all, File},
     io::Write,
-    sync::{Arc, Mutex},
+    sync::{Arc,Mutex,OnceLock},
 };
 
 mod cmd;
 
-pub const STATE_FILENAME: &str = ".window-state";
+pub const DEFAULT_STATE_FILENAME: &str = ".window-state";
+static STATE_FILENAME: OnceLock<String> = OnceLock::new();
+
+static USE_JSON: OnceLock<bool> = OnceLock::new();
```

Comment:
there is no need to use a static variable, you can just store it on the builder and clone/move it to the appropriate places, see https://github.com/tauri-apps/plugins-workspace/pull/546 for previous work.

Reply:
I restored the branch to latest dev and [added these changes](https://github.com/thewh1teagle/plugins-workspace/commits/feat/window-state/custom-state-filename/), I think you need to open the PR, 
If it's not possible I'll create another PR

Reply:
https://github.com/tauri-apps/plugins-workspace/pull/1138